docs/ocean/design_docs/vert-lagrang-remap/vert-lagrang-remap.rst

```diff
@@ -0,0 +1,540 @@
+compute the density and pressure fields based on the remapped ocean state and
+the diagnostic field :code:`vertVelocityTop` which is the vertical velocity 
+through the top of the layer. This is only used as a diagnostic variable for 
+computing the MOC streamfunction. None of the mixing parameterizations require
+a vertical velocity (Eulerian or diasurface velocity).
+
+Note: if `vertVelocityTop` is computed between regridding and remapping then it 
+can be computed as 
+
+.. code::
+   
+   vertVelocityTop(k) = vertVelocityTop(k+1) - div_hu(k) - 
+                        (layerThickness(k,tlev=2) - layerThickness(k,tlev=1))/dt
+
+If `vertVelocityTop` is computed after remapping, then :code:`div_hu` is no
+longer appropriate as it has been remapped. In this case, the Lagrangian layer 
+thickness should be stored in a scratch variable and then the vertical velocity 
+through the top of the layer can be computed:
+
+.. code::
+
+   layerThicknessALE = layerThickness(tlev=2)
+   
+   layerThickness(tlev=2) = layerThicknessTarget
+   
+   vertVelocityTop = (layerThickness(tlev=2) - layerThicknessALE)/dt
+
+The computation of :code:`vertTransportVelocityTop` and 
+:code:`vertGMBolusVelocitytop` is unchanged as these fields represent Eulerian 
+velocities.
+
+Implementation: Grid is sufficiently conditioned for accuracy and stability
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Date last modified: 2020/12/15
+
+Contributors: Darren Engwirda, Xylar Asay-Davis, Carolyn Begeman
+
+After determining the target grid, perform the following steps:
+
+#. Apply damping function to limit the rate of grid movement in one timestep 
+   [and enforce a maximum movement limit?]. 
+#. Optional: Assign :math:`h_k^{t+1}` to :math:`h_k^{lg}` if 
+   :math:`h_k^{t+1} - h_k^{lg}` is less than a minimum thickness alteration. 
+   This motivated by accuracy considerations, as each remapping may introduce 
+   errors. *Darren, would this improve PPR computational performance?*
+#. Smooth the layer interfaces to reduce horizontal gradients in layer 
+   thickness while conserving volume.
+#. Apply minimum layer thickness criterion. 
+
+Namelist options:
+
+- Minimum layer thickness
+- Maximum thickness change during remapping
+- Optional: minimum thickness change for remapping to occur
+- *Parameters in damping function*
+- *Parameters in smoothing function*
+
+Implementation: Ability to specify remapping method and its options
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Date last modified: 2020/12/15
+
+Contributors: Darren Engwirda, Xylar Asay-Davis, Carolyn Begeman
+
+Namelist options:
+
+- frequency with which remapping should be performed (on which timestep)
+- order of the remapping
+- order of edge slope estimates
+- monotone limiter
+- boundary condition option
+- option to output some diagnostics?
+- *Some other remapping options here*
```

Comment:
@dengwirda Feel free to contribute here

Reply:
I think you've covered all the options that PPR supports here. At this stage, let's not be overly specific about which options are the "right" ones to use. Hopefully initial comparisons against the flux-form variant will offer some guidance.